kani-compiler/src/kani_middle/reachability.rs

```diff
@@ -520,27 +530,43 @@ fn should_codegen_locally(instance: &Instance) -> bool {
     !instance.is_foreign_item()
 }
 
-fn collect_alloc_items(alloc_id: AllocId) -> Vec<MonoItem> {
+fn collect_alloc_items(tcx: TyCtxt, alloc_id: AllocId) -> Vec<MonoItem> {
     trace!(?alloc_id, "collect_alloc_items");
     let mut items = vec![];
     match GlobalAlloc::from(alloc_id) {
         GlobalAlloc::Static(def) => {
-            // This differ from rustc's collector since rustc does not include static from
-            // upstream crates.
-            let instance = Instance::try_from(CrateItem::from(def)).unwrap();
-            should_codegen_locally(&instance).then(|| items.push(MonoItem::from(def)));
+            let int_def_id = rustc_internal::internal(tcx, def.def_id());
+            let anonymous = match tcx.def_kind(int_def_id) {
+                rustc_hir::def::DefKind::Static { nested, .. } => nested,
+                _ => false,
+            };
+            if anonymous {
+                let alloc = def.eval_initializer().unwrap();
+                items.extend(
+                    alloc
+                        .provenance
+                        .ptrs
+                        .iter()
+                        .flat_map(|(_, prov)| collect_alloc_items(tcx, prov.0)),
+                );
+            } else {
+                // This differ from rustc's collector since rustc does not include static from
```

Comment:
```suggestion
                // This differs from rustc's collector since rustc does not include static from
```
(I know that this typo was in the prior code already.)